homeassistant/components/lutron_caseta/device_trigger.py

```diff
@@ -430,9 +430,11 @@ async def async_attach_trigger(
     """Attach a trigger."""
     device_registry = dr.async_get(hass)
     device = device_registry.async_get(config[CONF_DEVICE_ID])
+    assert device
```

Comment:
This seems like a bug. What if the device isn't available anymore, but still the automation is loaded and tries to attach to that device ID?

This is a great example of asserting should not be used.

Reply:
Would something like this be better?

```python
if not device:
    raise ValueError(f"Device {config[CONF_DEVICE_ID]} not found in registry")
```